packages/ui/src/components/input/select/index.tsx

```diff
@@ -121,15 +121,10 @@ function Component<T extends SelectOption<ValueType>>(
         setOpen(!open);
     }, [open]);
 
-    const handlePick = useCallback(
-        (event: SpecificMouseEvent<HTMLLIElement>) => {
-            if (!event.target) return;
-            const value = (event.target as HTMLLIElement).dataset.value;
-            if (!!onChange && !!value) onChange(JSON.parse(value) as T);
-            setOpen(false);
-        },
-        [onChange]
-    );
+    const getPickHandler = (option: T) => () => {
```

Comment:
Wondering if we want to keep the `useCallback`.

Reply:
Addressed in dd3e6094602f440f4ca5080639c5e074c342ad6d